### PR TITLE
PAS-207: XML Comments

### PR DESCRIPTION
**Pull request by jrmccannon**
Closes https://github.com/passwordless/passwordless-dotnet/issues/35

### CODE REVIEW

File: src/Passwordless/Models/Credential.cs [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+    /// <param name="attestationFmt"><see cref="AttestationFmt"/></param>
+    /// <param name="createdAt"><see cref="CreatedAt"/></param>
+    /// <param name="aaGuid"><see cref="AaGuid"/></param>
+    /// <param name="lastUsedAt"><see cref="LastUsedAt"/></param>
+    /// <param name="rpid"><see cref="RPID"/></param>
+    /// <param name="origin"><see cref="Origin"/></param>
+    /// <param name="country"><see cref="Country"/></param>
+    /// <param name="device"><see cref="Device"/></param>
+    /// <param name="nickname"><see cref="Nickname"/></param>
+    /// <param name="userId"><see cref="UserId"/></param>
```
Comment (Tyrrrz) [inline src/Passwordless/Models/Credential.cs:25]: We could also just not include the params in the docs if it makes things messy for no significant value

File: src/Passwordless/Models/AliasPointer.cs [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
+    /// </summary>
     public string UserId { get; }
+    /// <summary>
+    /// Hashed value of the user's alias
+    /// </summary>
     public string Alias { get; }
+    /// <summary>
+    /// Plaintext value of user's alias.
+    /// </summary>
     public string Plaintext { get; }
```
Comment (Tyrrrz) [inline src/Passwordless/Models/AliasPointer.cs:32]: Probably best to space these out now that there are docs

```suggestion
    /// <summary>
    /// Identifier for the user
    /// </summary>
    public string UserId { get; }

    /// <summary>
    /// Hashed value of the user's alias
    /// </summary>
    public string Alias { get; }

    /// <summary>
    /// Plaintext value of user's alias.
    /// </summary>
    public string Plaintext { get; }
```

File: src/Passwordless/Models/Credential.cs [hunk 3]
```diff
@@ -22,18 +43,63 @@ public class Credential
...
+    /// </summary>
     public string Device { get; }
+    /// <summary>
+    /// Friendly name for credential.
+    /// </summary>
     public string Nickname { get; }
+    /// <summary>
+    /// Identifier for the user
+    /// </summary>
     public string UserId { get; }
```
Comment (Tyrrrz) [inline src/Passwordless/Models/Credential.cs:104]: Probably best to space these out now that there are docs

```suggestion
    /// <summary>
    /// Descriptor of the credential as defined by the WebAuthn specification
    /// <see href="https://w3c.github.io/webauthn/#enumdef-publickeycredentialtype" />
    /// </summary>
    public CredentialDescriptor Descriptor { get; }

    /// <summary>
    /// Public key of the passkey pair.
    /// </summary>
    public byte[] PublicKey { get; }

    /// <summary>
    /// Byte array of user identifier
    /// </summary>
    public byte[] UserHandle { get; }

    /// <summary>
    /// Number of times credential has been used
    /// </summary>
    public uint SignatureCounter { get; }

    /// <summary>
    /// Attestation Statement format used to create credential
    /// </summary>
    public string AttestationFmt { get; }

    /// <summary>
    /// When the credential was created 
    /// </summary>
    public DateTime CreatedAt { get; }

    /// <summary>
    /// The AAGUID of the authenticator. Can be used to identify the make and model of the authenticator.
    /// <see href="https://www.w3.org/TR/webauthn/#aaguid"/>
    /// </summary>
    public Guid AaGuid { get; }

    /// <summary>
    /// Last time credential was used
    /// </summary>
    public DateTime LastUsedAt { get; }
    
    /// <summary>
    /// Relying Party identifier
    /// <see href="https://w3c.github.io/webauthn/#dom-publickeycredentialcreationoptions-rp"/>
    /// </summary>
    public string RPID { get; }
    
    /// <summary>
    /// Domain credential was created for
    /// </summary>
    public string Origin { get; }
    /// <summary>
    /// Optional country credential was created in
    /// </summary>
    public string Country { get; }
    
    /// <summary>
    /// Device the credential was created on
    /// </summary>
    public string Device { get; }
    
    /// <summary>
    /// Friendly name for credential.
    /// </summary>
    public string Nickname { get; }
    
    /// <summary>
    /// Identifier for the user
    /// </summary>
    public string UserId { get; }
```

**Review by Tyrrrz — APPROVED** [review]

**Comment by Tyrrrz** [comment]
Updated the issue description to link it to #35

**Comment by Tyrrrz** [comment]
Can this be reviewed or is there more to come?

File: src/Passwordless/Models/Credential.cs [hunk 4]
```diff
@@ -1,7 +1,14 @@
 ﻿namespace Passwordless;
 
+/// <summary>
+/// Store credential used to authenticate users. Follows WebAuthN specification
+/// <see href="https://www.w3.org/TR/webauthn-2/#public-key-credential"/>
```
Comment (abergs) [inline src/Passwordless/Models/Credential.cs:5]: I wouldn't say we actually follow the specification, we add stuff like nickname, device, country, userid etc.

I would say "A passkey credential stored by Passwordless"

File: src/Passwordless/Models/Credential.cs [hunk 5]
```diff
@@ -1,7 +1,14 @@
 ﻿namespace Passwordless;
 
+/// <summary>
+/// Store credential used to authenticate users. Follows WebAuthN specification
```
Comment (abergs) [inline src/Passwordless/Models/Credential.cs:4]:
```suggestion
/// The passkey credential stored by Passwordless
```

File: src/Passwordless/Models/Credential.cs [hunk 6]
```diff
@@ -22,18 +29,76 @@ public class Credential
...
+    /// </summary>
     public byte[] PublicKey { get; }
+
+    /// <summary>
+    /// Byte array of user identifier
+    /// </summary>
     public byte[] UserHandle { get; }
+
+    /// <summary>
+    /// Number of times credential has been used
```
Comment (abergs) [inline src/Passwordless/Models/Credential.cs:49]:
```suggestion
    /// WebAuthn SignatureCounter, used for anti forgery. 
```

Often you're correct, but Apple and some others use set unix time to this counter.

**Review by abergs — COMMENTED** [review]
I had some small nitpicks, otherwise it looks good :)

After those are adressed we're good to go.

**Pull request merged**